src/test/unit/services/CaseSelectionService.test.ts

```diff
@@ -0,0 +1,175 @@
+import axios, { AxiosResponse } from 'axios';
+
+import { CaseApiDataResponse } from '../../../main/definitions/api/caseApiResponse';
+import { CaseType, CaseWithId, YesOrNo } from '../../../main/definitions/case';
+import { ServiceErrors } from '../../../main/definitions/constants';
+import { CaseState } from '../../../main/definitions/definition';
+import {
+  getRedirectUrl,
+  getUserApplications,
+  getUserCasesByLastModified,
+} from '../../../main/services/CaseSelectionService';
+import * as caseService from '../../../main/services/CaseService';
+import { CaseApi } from '../../../main/services/CaseService';
+import { mockApplications } from '../mocks/mockApplications';
+import { mockAxiosError } from '../mocks/mockAxios';
+import { mockValidCaseWithId } from '../mocks/mockCaseWithId';
+import { mockRequest } from '../mocks/mockRequest';
+import { mockEnglishClaimTypesTranslations } from '../mocks/mockTranslations';
+
+jest.mock('axios');
+const getCaseApiClientMock = jest.spyOn(caseService, 'getCaseApi');
+
+describe('Get redirect url tests', () => {
+  const userCase = mockValidCaseWithId;
+  const languageParam = '?lng=en';
+  it('Should return claimant application link when user case state is awaiting submission to HMCTS', async () => {
+    userCase.state = CaseState.AWAITING_SUBMISSION_TO_HMCTS;
+    expect(getRedirectUrl(userCase, languageParam)).toEqual('/claimant-application/1234567890123456?lng=en');
+  });
+  it('Should return response hub link when user case state is not awaiting submission to HMCTS', async () => {
+    userCase.state = CaseState.ACCEPTED;
+    expect(getRedirectUrl(userCase, languageParam)).toEqual('/response-hub/1234567890123456?lng=en');
+  });
+});
+
+describe('Get user cases by last modified date tests', () => {
+  const req = mockRequest({});
+  const mockApiClient = {
+    createCase: jest.fn(),
+    getUserCases: jest.fn(),
+    downloadClaimPdf: jest.fn(),
+    updateDraftCase: jest.fn(),
+    getUserCase: jest.fn(),
+  };
+
+  afterEach(() => {
+    mockApiClient.getUserCases.mockClear();
+    mockApiClient.createCase.mockClear();
+    mockApiClient.downloadClaimPdf.mockClear();
+    mockApiClient.updateDraftCase.mockClear();
+    mockApiClient.getUserCase.mockClear();
+  });
+
+  it('Should Return user cases by last modified date', async () => {
+    const response: AxiosResponse<CaseApiDataResponse[]> = {
+      data: [
+        {
+          id: '12234',
+          state: CaseState.AWAITING_SUBMISSION_TO_HMCTS,
```

Comment:
if I'm being picky, the respondent wouldn't be able to see the case if it's in a draft state, only accepted & closed